src/app/clusters/device-energy-management-server/device-energy-management-server.cpp

```diff
@@ -629,7 +667,54 @@ void Instance::HandleModifyForecastRequest(HandlerContext & ctx, const Commands:
         return;
     }
 
-    status = mDelegate.ModifyForecastRequest(forecastId, slotAdjustments, adjustmentCause);
+    // Check the various values in the slot structures
+    auto iterator = slotAdjustments.begin();
+    while (iterator.Next())
```

Comment:
I don't see where we are checking status at the end of the list, so if the TLV in there is invalid we will happily successfully process things up to there and claim everything is good.  We need to be checking the status after the loop ends.

Reply:
I have added a check.